monorepo/services/server/src/routes/append-exclusions-to-campaign.js

```diff
@@ -0,0 +1,58 @@
+const { Router } = require('express');
+const multer = require('multer');
+const csvToJson = require('csvtojson');
+const createError = require('http-errors');
+const slug = require('slug');
+const asyncRoute = require('../utils/async-route');
+
+const router = Router();
+const dev = process.env.NODE_ENV === 'development';
+
+const storage = multer.diskStorage({
+  filename: (_, file, cb) => {
+    const uniqueSuffix = `${Date.now()}-${Math.round(Math.random() * 1E9)}`;
+    cb(null, `${file.fieldname}-${uniqueSuffix}`);
+  },
+});
+const upload = multer({ storage });
+
+router.post('/', upload.single('file'), asyncRoute(async (req, res) => {
+  const { file } = req;
+
+  const rows = await csvToJson().fromFile(file.path);
+  if (!rows || !rows.length) throw createError(400, 'Unable to find any rows in the provided file.');
+
+  let fieldName;
+  Object.keys(rows[0]).forEach((k) => {
+    const cleaned = slug(k.trim()).replace(/-/g, '');
+    if (/^suppress/.test(cleaned)) {
+      if (fieldName) throw createError(400, 'More than one suppress column was found the provided file.');
+      fieldName = k;
+    }
+  });
+  if (!fieldName) throw createError(400, 'Unable to find a suppress column in the provided file.');
+  const suppressions = rows.reduce((set, row) => {
+    const value = row[fieldName].trim().toLowerCase();
+    if (value) set.add(value);
```

Comment:
Will need to add additional filtering here. Notably:
- If not an email address (`x@x.x`), check for the following and remove:
- protocols (`[http|https|mailto]://...`)
- Leading `www.` on domains
- Trailing URL parts (`/...`)